### PR TITLE
Site Settings in SwiftUI: Related Posts

### PR DESCRIPTION
**Pull request by kean**
HACK week: update settings screen to use SwiftUI

- Replace `RelatedPostsSettingsViewController` (Objective-C) and add `SiteSettingsRelatedPostsView` (SwiftUI)
- Contains the save fix ([#20894](https://github.com/wordpress-mobile/WordPress-iOS/pull/20894))
- Fixes the layout issues (clipped views)

Before → After

settings-issues settings-issues-fixed

## To test:

- Open Site Settings / Related Posts
- Update the settings
- Verify that it matches the previous behavior (and addresses layout issues)
- Verify that after changing settings, it's immediately updated on the site

## Regression Notes
1. Potential unintended areas of impact: Site Settings / Related Posts
2. What I did to test those areas of impact (or what existing automated tests I relied on): manual testing
3. What automated tests I added (or what prevented me from doing so): n/a

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [ ] VoiceOver.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout) - [Evidence](https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/b1e31c6e-8f56-44dd-822d-1978fb9bfb79)
- [x] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Review by momo-ozawa — APPROVED** [review]
Works as described!

File: WordPress/Classes/ViewRelated/Blog/Site Settings/SiteSettingsRelatedPostsView.swift [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    ]
+}
+
+private extension RelatedPostsSettingsView {
+    enum Strings {
+        static let title = NSLocalizedString("relatedPostsSettings.title", value: "Related Posts", comment: "Title for screen that allows configuration of your blog/site related posts settings.")
+        static let showRelatedPosts = NSLocalizedString("relatedPostsSettings.showRelatedPosts", value: "Show Related Posts", comment: "Label for configuration switch to enable/disable related posts")
+        static let showHeader = NSLocalizedString("relatedPostsSettings.showHeader", value: "Show Header", comment: "Label for configuration switch to show/hide the header for the related posts section")
+        static let showThumbnail = NSLocalizedString("relatedPostsSettings.showThumbnail", value: "Show Images", comment: "Label for configuration switch to show/hide images thumbnail for the related posts")
+        static let optionsFooter = NSLocalizedString("relatedPostsSettings.optionsFooter", value: "Related Posts displays relevant content from your site below your posts", comment: "Information of what related post are and how they are presented")
```
Comment (dcalhoun) [inline WordPress/Classes/ViewRelated/Blog/Site Settings/SiteSettingsRelatedPostsView.swift:134]: 👋🏻 @kean I noticed [a translation inquiry](https://wordpress.slack.com/archives/C02RQC4LY/p1689091019847819) in the WordPress Slack regarding Related Posts. Specifically combining plural "Related Posts" with "displays," i.e Related Posts **displays**. Do you have thoughts on whether this string should be adjusted?

Comment (kean) [reply WordPress/Classes/ViewRelated/Blog/Site Settings/SiteSettingsRelatedPostsView.swift:134]: Hey, @dcalhoun. I don't think I'm on this Slack. I would appreciate it if you could reply.

The value was copied from the original version. I simply added the proper key.

```swift
return NSLocalizedString(@"Related Posts displays relevant content from your site below your posts", @"Information of what related post are and how they are presented");;
```

I think it means "Related Posts (section) displays" or something like that, but "display" sounds more natural in this context.

Comment (kean) [reply WordPress/Classes/ViewRelated/Blog/Site Settings/SiteSettingsRelatedPostsView.swift:134]: Please, disregard my comment. I created the account and replied directly in the thread.

**Pull request merged**